### ISSUE TITLE
Link an Azure Automation Account to a Log Analytics workspace

### ISSUE THREAD

**Issue by jseerden** [terraform-providers/terraform-provider-azurerm#2080]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Using Terraform to deploy Azure Automation Accounts, Log Analytics workspaces and the AzureAutomation solution. I am only missing to ability to link an Azure Automation account to a Log Analytic workspace containing the AzureAutomation solution.

### New or Affected Resource(s)

* https://www.terraform.io/docs/providers/azurerm/r/log_analytics_solution.html

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

To my knowledge, linking an Azure Automation account is possible only using ARM templates: https://github.com/MicrosoftDocs/azure-docs/issues/6537

* #0000

**Comment by draggeta** [comment]
This resource can be deployed via the API and it's available in the Go Azure SDK from the preview/operationalinsights/mgmt/2015-11-01-preview version.

**Comment by tombuildsstuff** [comment]
hi @jseerden @draggeta 

Just to let you know that this has been released as a part of v1.18 of the AzureRM Provider ([the full changelog is available here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.18.0/CHANGELOG.md)). You can upgrade to this by specifying the version in the provider block (as shown below) and then running `terraform init -upgrade`

```
provider "azurerm" {
  version = "=1.18.0"
}
```

Thanks!
